service/qos/qos_common.hh

```diff
@@ -30,6 +30,8 @@ namespace qos {
 
 enum class include_effective_names { yes, no };
 
+enum class internal_query { yes, no, default_value };
```

Comment:
Why do we need an additional `default_value` variant? Wouldn't just `yes` and `no` suffice?

Reply:
My thinking was that we have the group0 context, the user command context, and other contexts where the distinction is not important. Perhaps there could be internal calls where we want to keep the timeout short, like in the legacy cache update.
And it was also convenient to have a default value and not having to classify all possible paths.
Maybe the name `internal_query` can be changed to something more clear, like `from_raft`?

Reply:
> My thinking was that we have the group0 context, the user command context, and other contexts where the distinction is not important.

So maybe the enum should be called `query_context` and the variants should be `group0`, `user`, `unspecified` (or something like that)?

My main problem with "default value" is that it is not clear what it means next to "yes" and "no". The enum seems to answer the question whether the query is internal or not - "yes" and "no" are clear, but "default value" is not.

`from_raft` also sounds OK to me. I suspect that it will become unnecessary if we switch service levels to read from the cache, so future proofing the enum in the sense of being able to add more values to it in the future might be not necessary.

Reply:
I renamed the enum to `query_context {group0, user, unspecified}`